### DISCUSSION TITLE
What link do i have to put in to connect my drive of another account?

### DISCUSSION THREAD

**Discussion by ryadios** [TheLastBen/fast-stable-diffusion#1932] [Q&A]
So there is a ``Shared_Drive`` section, i wanted to know what type of link does it take? A shared folder link doesn't work as i have tried

**Comment by TheLastBen** [comment]
you just need to use the other account with colab, the Shared Drive is meant for this case : https://support.google.com/a/users/answer/7212025?hl=en#:~:text=Shared%20drives%20are%20special%20folders,team%20instead%20of%20an%20individual.

**Comment by Omenizer** [comment]
But for Shared Drive I need a work or school account and have to pay, right? No way to make on with a free acct

**Reply by TheLastBen** [reply]
It's for schools and businesses

**Reply by Omenizer** [reply]
Do you think it would be possible to add an option to install to a shared *folder*? That would be very practical and save a lot of space across several accounts, especially with the controlnet models and upscaler networks taking 2/3 of a free drive already.

My path to model is a shared folder, and this works really well. Loras shared too.

**Reply by TheLastBen** [reply]
You can do that manually because that requires more than one simple operation

**Comment by Mossbraker** [comment]
If you want to connect the drive of a second Google account, [follow the steps below:](https://stackoverflow.com/questions/53728127/mount-multiple-drives-in-google-colab)

1\. Mount your first drive as usual, by running the "Connect Google Drive" cell.

2\. Run the below code:
```python
!sudo add-apt-repository -y ppa:alessandro-strada/ppa 2>&1 > /dev/null
!sudo apt-get update -qq 2>&1 > /dev/null
!sudo apt -y install -qq google-drive-ocamlfuse 2>&1 > /dev/null
!google-drive-ocamlfuse
```

3\. Allow gdfuse access your second Google account.

The output from the previous step should include an error like this:
>Failure("Error opening URL:https://accounts.google.com/o/oauth2/auth?client_id=... ")

![image](https://user-images.githubusercontent.com/37157218/230566016-2281ca45-8564-4d23-acde-eb88d2fa0045.png)
Click on the link in the error message and authenticate your account.

4\. Run the below code:

```python
!sudo apt-get install -qq w3m # to act as web browser 
!xdg-settings set default-web-browser w3m.desktop # to set default browser
%cd /content
!mkdir drive
%cd drive
!mkdir MyDrive2
%cd ..
%cd ..
!google-drive-ocamlfuse /content/drive/MyDrive2
```
After this step, your second drive will be mounted to _/content/drive/MyDrive2_

**Reply by Omenizer** [reply]
I have a few accounts connected via shared folders, no code needed - so I have access to the Models, Lora and Embeds from all accounts. 

What I want to do is have a single drive with the SD install so I don't have to update/install extensions and synch settings

**Reply by Mossbraker** [reply]
Ah, yeah that's mostly what I do too. But I have a school account with unlimited storage, so sometimes if I'm training and saving a bunch of ckpts or something that requires lots of storage (and if I've already used up all my free Colab time on my school account) it's nice to be able to connect my school account just for storage.
>What I want to do is have a single drive with the SD install so I don't have to update/install extensions and synch settings

You can do that manually, same as you probably would have done with models and Loras. 

To accomplish what you want:
1\. Share the 'sd' folder from the root of your 'main' Google Drive account with your 'other' accounts. 

2\. On your 'other' accounts:
\- Go to Google Drive and if there's already an 'sd' folder on the root of your Drive move, rename, or delete it. If you delete it, obviously make sure you have backups of any output, models, config files, etc. that you don't want to lose.
\- Go to the [shared with me](https://drive.google.com/drive/shared-with-me) section in Google Drive (if you're signed into multiple accounts, this link will take you to your primary account and you'll have to switch accounts and navigate to 'Shared with me' using the sidebar).
\- Right click on the 'sd' folder that was shared from your 'main' account and select 'Add shortcut to Drive' > 'My Drive' > 'Add Shortcut' to add a shortcut to your drive.

This will add a shortcut to your 'other' Drive account which will function the same in Colab as the folder `/content/gdrive/MyDrive/sd/` and so you'll only have to maintain one copy of everything.
